sentinelsat/sentinel.py

```diff
@@ -572,11 +572,11 @@ def download(self, id, directory_path=".", checksum=True, **kwargs):
 
 
         .. versionchanged:: 0.15
-           Added ``**kwargs`` parameter to allow easier specialization of the :class:`SentinelAPI` cass.
+           Added ``**kwargs`` parameter to allow easier specialization of the :class:`SentinelAPI` class.
         """
         product_info = self.get_product_odata(id)
-        ext = kwargs.get("file_ext", ".zip")
-        path = join(directory_path, product_info["title"] + ext)
+        filename = self._get_filename(product_info)
+        path = join(directory_path, filename)
```

Comment:
OK, this will require a change in the PR #441 "New specialized client for the GNSS end-point on scihub"